core/src/Kokkos_View.hpp

```diff
@@ -1548,6 +1609,18 @@ class View : public ViewTraits<DataType, Properties...> {
 #endif
     //------------------------------------------------------------
 
+    if (check_input_args) {
+#ifdef KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HOST
```

Comment:
this is not necessary: the consturctor is never called on the device. Note it is not marked as a KOKKOS_FUNCTION

Reply:
Removed.